vertical-pod-autoscaler/enhancements/7862-cpu-startup-boost/README.md

```diff
@@ -0,0 +1,148 @@
+
+1. The VPA Updater monitors pods targeted by the VPA object and when the pod
+condition is `Ready`, it scales down the CPU resources to the appropriate
+non-boosted value: `max(existing VPA recommendation for that container, minimum configured value)`.
+    * The scale down is applied [in-place](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources).
+
+### API Changes
+
+To extend [`ContainerResourcePolicy`](https://github.com/kubernetes/autoscaler/blob/0a34bf5d3a71b486bdaa440f1af7f8d50dc8e391/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L190) with a `StartupBoost.CPU.Factor` field, that will instruct VPA the factor by which to multiply the initial resource request and limit of the containers' targeted by the VPA object.
+
+This AEP also proposes to extend [`ContainerScalingMode`](https://github.com/kubernetes/autoscaler/blob/0a34bf5d3a71b486bdaa440f1af7f8d50dc8e391/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L230-L235) with a new `StartupBoostOnly` mode to allow users to only enable the startup boost feature and not vanilla VPA altogether.
+
+
+#### Priority of `StartupBoost`
+
+The new `StartupBoost` field will take precedence over the rest of the container resource policy configurations. Functioning independently from all other fields in [`ContainerResourcePolicy`](https://github.com/kubernetes/autoscaler/blob/0a34bf5d3a71b486bdaa440f1af7f8d50dc8e391/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L190), except for [`ContainerName`](https://github.com/kubernetes/autoscaler/blob/0a34bf5d3a71b486bdaa440f1af7f8d50dc8e391/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L191-L194) and [`Mode`](https://github.com/kubernetes/autoscaler/blob/0a34bf5d3a71b486bdaa440f1af7f8d50dc8e391/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L195-L197). This means that a container's CPU request/limit can be boosted during startup beyond [`MaxAllowed`](https://github.com/kubernetes/autoscaler/blob/0a34bf5d3a71b486bdaa440f1af7f8d50dc8e391/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L202-L205), for example, or it will be able to be boosted even if
+CPU is  explicitly excluded from [`ControlledResources`](https://github.com/kubernetes/autoscaler/blob/0a34bf5d3a71b486bdaa440f1af7f8d50dc8e391/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L207-L211).
+
+#### CPU startup boost and VPA autoscaling modes
+
+Initially, CPU startup boost will only be able to be used if the container or VPA scaling mode is [`InPlaceOrRecreate`](https://github.com/kubernetes/autoscaler/blob/0a34bf5d3a71b486bdaa440f1af7f8d50dc8e391/vertical-pod-autoscaler/enhancements/4016-in-place-updates-support/README.md?plain=1#L61), since it does not make sense to use this feature with disruptive modes of VPA
+
+Here's an example of the VPA CRD, incorporating CPU boosting (only shows relevant sections):
```

Comment:
Can we also show an example of a VPA configuration where the user might want to use VPA normally AND the boost feature together?